src/transformers/trainer_seq2seq.py

```diff
@@ -254,7 +257,15 @@ def prediction_step(
         # XXX: adapt synced_gpus for fairscale as well
         # Priority (handled in generate):
         # gen_kwargs > model.generation_config > default GenerationConfig()
-        gen_kwargs = self._gen_kwargs.copy()
+
+        if gen_kwargs:
+            # overriding gen_kwargs have been provided, use those
+            gen_kwargs = gen_kwargs.copy()
+        elif hasattr(self, "_gen_kwargs"):
+            gen_kwargs = self._gen_kwargs.copy()
+        else:
+            gen_kwargs = {}
```

Comment:
```suggestion
        if len(gen_kwargs) == 0 and hasattr(self, "_gen_kwargs"):
            gen_kwargs = self._gen_kwargs.copy()
```
Might be simpler to just do this.

Reply:
Thanks for super quick response. This would certainly be simpler. I guess we don't need to do a copy of `gen_kwargs` here since `prediction_step` is only going to run on 1 instance of data, is that accurate?

Reply:
The lines below change the dictionary, so it's better to be safe.

Reply:
True, I was wondering if we should do:
```python
if len(gen_kwargs) == 0 and hasattr(self, "_gen_kwargs"):
       gen_kwargs = self._gen_kwargs.copy()
else:
       gen_kwargs = gen_kwargs.copy()
```

or last 2 lines are not required ?